torch/distributed/_shard/sharded_tensor/_ops/_common.py

```diff
@@ -378,3 +378,23 @@ def _handle_max_norm_col_wise(
         .contiguous()
     )
     return local_shard_norm_renormed, gathered_inputs
+
+
+def narrow_tensor(tensor, metadata):
+    """
+    narrow the tensor according to the metadata
+    """
+    narrowed_tensor = tensor
+    shard_offsets = metadata.shard_offsets
+    shard_sizes = metadata.shard_sizes
+    for idx, (offset, size) in enumerate(zip(shard_offsets, shard_sizes)):
+        if size < tensor.size(idx):
+            # Reshape to get shard for this rank and we don't want autograd
+            # recording here for the narrow op and 'local_shard' should be a
+            # leaf variable in the autograd graph.
+            narrowed_tensor = narrowed_tensor.narrow(
+                idx,
+                shard_offsets[idx],
+                shard_sizes[idx]
+            )
+    return narrowed_tensor
```

Comment:
Don't we do something similar in `_shard_parameter`/`_shard_tensor`? Maybe we can consolidate the code to use this method?